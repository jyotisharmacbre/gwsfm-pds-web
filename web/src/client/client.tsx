import axios from 'axios';
import authentication from '@kdpw/msal-b2c-react';
import appConfig from '../helpers/config-helper';

const config = appConfig();
export const baseURL = config.REACT_APP_MIDDLETIER_URL;
export const baseAPI = axios.create({ baseURL });
export const userServiceURL = config.REACT_APP_GATEWAY_URL;
export const userServiceAPI = axios.create({ baseURL: userServiceURL });

const isTokenHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('tokenHandlerEnabled') &&
    !config['tokenHandlerEnabled']
    ? false
    : true;
};

const isErrorHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('errorHandlerEnabled') &&
    !config['errorHandlerEnabled']
    ? false
    : true;
};

const isSuccessHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('successHandlerEnabled') &&
    !config['successHandlerEnabled']
    ? false
    : true;
};

const requestHandler = request => {
  if (isTokenHandlerEnabled(request)) {
    // Add request token here
    const token = authentication.getAccessToken();
    request.headers['accessToken'] = token;
  }
  return request;
};

baseAPI.interceptors.request.use(request => requestHandler(request));
userServiceAPI.interceptors.request.use(request => requestHandler(request));

const errorHandler = error => {
  if (isErrorHandlerEnabled(error.config)) {
    // Handle errors
  }
  return Promise.reject({ ...error });
};

const successHandler = response => {
  if (isSuccessHandlerEnabled(response.config)) {
    // Handle responses
  }
  return response;
};

baseAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);

userServiceAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);
