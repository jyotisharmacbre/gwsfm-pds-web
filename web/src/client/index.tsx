export { baseAPI, userServiceAPI} from './client';
