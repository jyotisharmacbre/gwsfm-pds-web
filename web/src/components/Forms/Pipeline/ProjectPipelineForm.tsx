import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { IState } from '../../../store/state';
import { getLookupDescription } from '../../../helpers/utility-helper';
import { LookupItems } from '../../../helpers/constants';
import moment from 'moment';
import { Link } from 'react-router-dom';
import GridTable from '../../Table/GridTable';
interface Props {
  pipelineValues: any;
  lookupValues: any;
}
const ProjectPipelineForm: React.FC<Props> = (props: any) => {
  const { pipelineValues, lookupValues } = props;
  const getPipelineValues = allLookups => {
    let data = pipelineValues.map(function(rowProject) {
      var statusID = rowProject.status;
      if (!isNaN(statusID) && allLookups.length > 0)
        rowProject.status = getLookupDescription(
          allLookups,
          rowProject.status,
          LookupItems.Project_Status
        );

      var contractID = rowProject.contractTypeId;
      if (contractID > 0 && allLookups.length > 0)
        rowProject.contractTypeId = getLookupDescription(
          allLookups,
          rowProject.contractTypeId,
          LookupItems.ContractType
        );
      rowProject.lastModified = moment(rowProject.lastModified).format(
        'MM/DD/YYYY'
      );
      rowProject.commenceDate =
        rowProject.commenceDate != ''
          ? moment(rowProject.commenceDate).format('MM/DD/YYYY')
          : '';

      rowProject.cdmNotifiable = rowProject.cdmNotifiable ? 'Yes' : 'No';

      rowProject.name = (
        <Link
          to={{
            pathname: `/projectoverview/${rowProject.projectId}`,
            state: { fromPipeline: true }
          }}
        >
          {rowProject.name}
        </Link>
      );
      return rowProject;
    });
    return data;
  };
  return (
    <React.Fragment>
      <GridTable
        columns={getTableColumns()}
        data={getPipelineValues(lookupValues)}
        ActionList={[]}
      />
    </React.Fragment>
  );
};
const getTableColumns = () => {
  return [
    {
      title: 'Project name',
      field: 'name',
      customFilterAndSearch: (term: any, rowData: any) =>
        (term = rowData.name.length)
    },
    { title: 'Owner', field: 'projectOwner' },
    { title: 'Last Update', field: 'lastModified' },
    {
      title: 'Client/customer',
      field: 'contractorId'
    },
    {
      title: 'Prob of winning',
      field: 'probabilityOfWinning'
    },
    {
      title: 'Status',
      field: 'status'
    },
    {
      title: 'Expected start date',
      field: 'commenceDate',
      type: 'date'
    },
    {
      title: 'Approx value',
      field: 'approxValue'
    },
    {
      title: 'Contract type',
      field: 'contractTypeId'
    },
    {
      title: 'CMD notifiable',
      field: 'cdmNotifiable'
    },
    {
      title: 'Sold margin',
      field: 'soldmargin'
    },
    {
      title: 'Weighted TCV',
      field: 'weightedTCV'
    },
    {
      title: 'Rank',
      field: 'rank'
    }
  ];
};

const mapStateToProps = (state: IState) => ({
  initialValues: state.pipelineGrid.pipelineDetails
});
export default connect(mapStateToProps)(ProjectPipelineForm);
