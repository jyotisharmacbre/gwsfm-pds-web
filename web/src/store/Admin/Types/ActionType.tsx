export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
