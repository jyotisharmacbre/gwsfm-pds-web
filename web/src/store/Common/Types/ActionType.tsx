export enum ActionType {
  DEFAULT = 'DEFAULT'
}
