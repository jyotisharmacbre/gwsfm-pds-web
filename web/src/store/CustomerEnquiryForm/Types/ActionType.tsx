export enum ActionType {
  DEFAULT = 'DEFAULT',
  PROJECT_ADD = 'PROJECT_ADD',
  PROJECT_ADD_ERROR = 'PROJECT_ADD_ERROR',
  PROJECT_EDIT_SUCCESS = 'PROJECT_EDIT_SUCCESS',
  GET_ENQUIRY_OVERVIEW_SUCCESS = 'GET_ENQUIRY_OVERVIEW_SUCCESS',
  GET_ENQUIRY_OVERVIEW_ERROR = 'GET_ENQUIRY_OVERVIEW_ERROR',
  GET_PROJECT_DETAIL_SUCCESS = 'GET_PROJECT_DETAIL_SUCCESS',
  GET_PROJECT_DETAIL_ERROR = 'GET_PROJECT_DETAIL_ERROR',
  RESET_PROJECT_DETAIL_STATE = 'RESET_PROJECT_DETAIL_STATE',
  SET_PROJECT_ID_STATE = 'SET_PROJECT_ID_STATE'
}
