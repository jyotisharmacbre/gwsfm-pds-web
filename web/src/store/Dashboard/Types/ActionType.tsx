export enum ActionType {
  PROJECT_DASHBOARD_GRID_ERROR = 'PROJECT_DASHBOARD_GRID_ERROR',
  PROJECT_DASHBOARD_GRID_DETAILS = 'PROJECT_DASHBOARD_GRID_DETAILS'
}
