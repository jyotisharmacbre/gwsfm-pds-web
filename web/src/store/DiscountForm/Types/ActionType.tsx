export enum ActionType {
	DISCOUNT_FORM_DATA_ADD = 'DISCOUNT_FORM_DATA_ADD',
	DISCOUNT_FORM_DATA_ERROR = 'DISCOUNT_FORM_DATA_ERROR',
	DISCOUNT_FORM_DATA_EDIT = 'DISCOUNT_FORM_DATA_EDIT',
	RESET_DISCOUNT_FORM_STATE = 'RESET_DISCOUNT_FORM_STATE',
	RESET_DISCOUNT_FORM_STATE_NOTIFIER = 'RESET_DISCOUNT_FORM_STATE_NOTIFIER',
	DISCOUNT_FORM_DATA_GET = 'DISCOUNT_FORM_DATA_GET',
	DISCOUNT_FORM_DATA_GET_ERROR = 'DISCOUNT_FORM_DATA_GET_ERROR'
}
