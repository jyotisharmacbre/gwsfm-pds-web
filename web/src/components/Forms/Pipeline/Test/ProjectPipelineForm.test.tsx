import React from 'react';
import nock from 'nock';
import { baseURL, userServiceURL } from '../../../../client/client';
import { getUsersEmailData, pipelineGridData, intialLookupvalues, customerContractList } from './ProjectPipelineFormTestData';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import { MemoryRouter } from 'react-router';
import ProjectPipelineForm from '../ProjectPipelineForm';
import { ActionType } from '../../../../store/pipeline/Types/ActionType';
import projectPipelineDetailReducer, { initialState } from '../../../../store/pipeline/Reducer';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);
nock(baseURL)
  .get('/api/Projects/Getall')
  .reply(200, pipelineGridData);

nock(userServiceURL)
  .post('/api/users/getusernamesforemailids')
  .reply(200, getUsersEmailData);
let store;
const setUpStore = (pipelineGridData) => {
  store = mockStore({
    pipelineGrid: pipelineGridData
  });
};
let props = {
  pipelineValues: pipelineGridData,
  lookupValues: intialLookupvalues,
  userNamesForEmailsValues: getUsersEmailData,
  contractCustomerList: customerContractList
};

describe('Dashboard Form testCases', () => {
  let wrapper: any;

  const componentMount = props => {
    wrapper = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
          <MemoryRouter>
            <ProjectPipelineForm {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>
    );
  };

  it('Defines the component', () => {
    setUpStore(pipelineGridData);
    componentMount({ props });
    expect(wrapper).toBeDefined();
  });
  it('should match the snapshot', () => {
    setUpStore(pipelineGridData);
    componentMount({ props });
    expect(wrapper).toMatchSnapshot();
  });
});
describe('Dashboard form reducer', () => {
  it('should handle Get PROJECT DASHBOARD GRID DETAILS successfully', () => {
    const projectPipelineGridAction: any = {
      type: ActionType.PROJECT_PIPELINE_GRID_DETAILS
    };
    expect(
      projectPipelineDetailReducer(initialState, projectPipelineGridAction)
    ).toMatchSnapshot();
  });
});