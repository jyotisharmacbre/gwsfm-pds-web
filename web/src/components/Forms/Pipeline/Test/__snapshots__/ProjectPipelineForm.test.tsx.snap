// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Form testCases should match the snapshot 1`] = `ReactWrapper {}`;

exports[`Dashboard form reducer should handle Get PROJECT DASHBOARD GRID DETAILS successfully 1`] = `
Object {
  "data": undefined,
  "error": null,
  "loading": false,
  "pipelineDetails": undefined,
  "projectChartSummary": Object {
    "data": Array [],
    "error": null,
    "loading": false,
  },
  "totalNumberOfRecord": undefined,
}
`;
